### PR TITLE
Implement metadata GET deviceservice/deviceprofile and PATCH deviceservice

### PR DESCRIPTION
**Pull request by jinlinGuan**
Fix https://github.com/edgexfoundry/edgex-taf/issues/209
Also correct filenames in PUT-Negative-UploadFile.robot

Signed-off-by: Ginny Guan <ginny@iotechsys.com>

### CODE REVIEW

**Comment by jinlinGuan** [comment]
failed testcases: `patch device service with empty baseAddress` is expected to return 400 but 200

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/GET-Positive.robot [hunk 1]
```diff
@@ -1,90 +1,109 @@
...
-    And Should Have Content-Type "application/json"
-    And Validate Response Schema
-    And Response Time Should Be Less Than "1200"ms
-    [Teardown]  Delete Device Profiles
+    Given Generate An Device Profile Sample  Test-Profile-1
+    And Create Device Profile ${deviceProfile}
+    When Query Device Profile By Name  Test-Profile-1
+    Then Should Return Status Code "200" And profile
+    And Should Return Content-Type "application/json"
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
```
Comment (cherrycl) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/GET-Positive.robot:62]: Suggest validating the response contains the correct profile name.

Comment (jinlinGuan) [reply TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/GET-Positive.robot]: Added the validations to device profile and devices ervice
https://github.com/edgexfoundry/edgex-taf/blob/ead9e9a1d381178856840a022e04d9d054862201/TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceservice/GET-Positive.robot#L67
https://github.com/edgexfoundry/edgex-taf/blob/ead9e9a1d381178856840a022e04d9d054862201/TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/GET-Positive.robot#L61

**Review by cherrycl — APPROVED** [review]
LGTM

**Pull request merged**